js_modules/dagster-ui/packages/ui-core/src/asset-data/AssetStaleStatusDataProvider.tsx

```diff
@@ -0,0 +1,80 @@
+import {ApolloClient, gql, useApolloClient} from '@apollo/client';
+import React from 'react';
+
+import {
+  AssetStaleStatusDataQuery,
+  AssetStaleStatusDataQueryVariables,
+} from './types/AssetStaleStatusDataProvider.types';
+import {tokenForAssetKey, tokenToAssetKey} from '../asset-graph/Utils';
+import {AssetKeyInput} from '../graphql/types';
+import {liveDataFactory} from '../live-data-provider/Factory';
+import {LiveDataThreadID} from '../live-data-provider/LiveDataThread';
+
+function init() {
+  return liveDataFactory(
+    () => {
+      return useApolloClient();
+    },
+    async (keys, client: ApolloClient<any>) => {
+      const {data} = await client.query<
+        AssetStaleStatusDataQuery,
+        AssetStaleStatusDataQueryVariables
+      >({
+        query: ASSET_STALE_STATUS_QUERY,
+        fetchPolicy: 'network-only',
+        variables: {
+          assetKeys: keys.map(tokenToAssetKey),
+        },
+      });
+      return Object.fromEntries(
+        data.assetNodes.map((node) => [tokenForAssetKey(node.assetKey), node]),
```

Comment:
Overall I think this approach of making two hooks and two separate caches, and composing them to create the old unified hook is good, but it locks us in to this very specific separation of staleStatus vs. other status, and combined with the "threading" concept it feels like we're approaching peak complexity.  The guts of all this code use template types already, and I'm curious what it'd look like to make the query part of the configuration.  (Templating rather than composing?)

Related to my earlier comment though -- this assumes that stale is the expensive part now and the base live data is cheap!

It feels like we're very close to being able to say:

```
const data = useAssetLiveData<T>(assetKeys, ASSET_STALE_STATUS_QUERY)
```

Normally we can't pass the query as a variant because of the useQuery / hook rules, but in this case we're making the query imperatively, so it'd be fine to swap queries out. I bet most of the downstream code that assembles + caches the data would be ok with any query as long as the response contained `assetNodes: {assetKey: AssetKey}`

Doing it in this very uniform way offers some benefits I think, namely that the caches can be shared across pages easily...  but you're definitely right that the partial caching in both places may make it weird. The asset "stale" tags may flicker in / out after the primary state changes. 

As long as we're confident we only want to slice on staleStatus, maybe this current approach is better, but I'm not totally sure?

Reply:
We want to explicitly fetch them in separate queries for the parallelization.